src/sections/trade/TradePage.tsx

```diff
@@ -1,27 +1,63 @@
 import { Page } from "components/Layout/Page/Page"
 
 import * as React from "react"
-import { createComponent } from "@lit-labs/react"
-import { TradeSpa as LitTradeSpa } from "@galacticcouncil/trade-app"
+import { createComponent, EventName } from "@lit-labs/react"
+import { TradeApp as LitTradeApp } from "@galacticcouncil/trade-app"
 import { useAccountStore } from "state/store"
+import { MakeGenerics, useSearch } from "@tanstack/react-location"
+import { z } from "zod"
+import { Spinner } from "components/Spinner/Spinner.styled"
 
-export const TradeSpa = createComponent({
-  tagName: "gc-trade-spa",
-  elementClass: LitTradeSpa,
+export const TradeApp = createComponent({
+  tagName: "gc-trade-app",
+  elementClass: LitTradeApp,
   react: React,
+  events: {
+    onInit: "gc:init" as EventName<CustomEvent>,
+  },
 })
 
+const TradeAppSearch = z.object({
+  type: z.union([z.literal("assetIn"), z.literal("assetOut")]),
+  id: z.number().transform((value) => String(value)),
+})
+
+type SearchGenerics = MakeGenerics<{
+  Search: z.infer<typeof TradeAppSearch>
+}>
+
 export function TradePage() {
   const { account } = useAccountStore()
 
+  const search = useSearch<SearchGenerics>()
+
+  const [loaded, setLoaded] = React.useState(false)
+  const ref = React.useRef<LitTradeApp>(null)
```

Comment:
mega nitpick (klidne skip) 
```suggestion
  const [loaded, setLoaded] = useState(false)
  const ref = useRef<LitTradeApp>(null)
```

Reply:
jj, problém je ten, že musím kvůli `@lit-labs/react` importovat celý React takto:

```ts
import * as React from "react"
```

Dvojí import by také šel, ale editor bude vždy upřednostňovat `React`

Reply:
ah 😄 okej